src/unix/newlib/mod.rs

```diff
@@ -10,30 +10,58 @@ cfg_if! {
 }
 
 cfg_if! {
-    if #[cfg(any(target_os = "espidf"))] {
+    if #[cfg(target_os = "horizon")] {
+        pub type dev_t = u32;
+        pub type off_t = i64;
+    } else {
         pub type dev_t = ::c_short;
-        pub type ino_t = ::c_ushort;
         pub type off_t = ::c_long;
-    } else if #[cfg(any(target_os = "vita"))] {
-        pub type dev_t = ::c_short;
-        pub type ino_t = ::c_ushort;
-        pub type off_t = ::c_int;
-    } else {
-        pub type dev_t = u32;
+    }
+}
+
+cfg_if! {
+    if #[cfg(target_os = "horizon")] {
         pub type ino_t = u32;
-        pub type off_t = i64;
+    } else if #[cfg(target_arch = "sparc")] {
+        pub type ino_t = ::c_ulong;
+    } else {
+        pub type ino_t = ::c_ushort;
     }
 }
 
 pub type fsblkcnt_t = u64;
 pub type fsfilcnt_t = u32;
 pub type id_t = u32;
-pub type key_t = ::c_int;
+
+cfg_if! {
+    if #[cfg(target_os = "horizon")] {
+        pub type key_t = ::c_int;
+    } else {
+        pub type key_t = ::c_long;
+    }
+}
+
 pub type loff_t = ::c_longlong;
-pub type mode_t = ::c_uint;
-pub type nfds_t = u32;
+
+cfg_if! {
+    if #[cfg(target_os = "horizon")] {
+        pub type mode_t = ::c_uint;
+        pub type nfds_t = u32;
+    } else {
+        pub type mode_t = u32;
```

Comment:
Ditto for `mode_t` and `nfds_t`?

Reply:
`nfds_t`:

https://sourceware.org/git/?p=newlib-cygwin.git;a=blob;f=newlib/libc/sys/rtems/include/sys/poll.h;h=cc6ad49db033dfc63714d5062da30bf08b364dc3;hb=9e09d6ed83cce4777a5950412647ccc603040409#l42

Not sure about newlib structure of rtems, but i found it only there.